bsd/acl.pyx

```diff
@@ -259,33 +259,30 @@ cdef class ACLEntry(object):
         return {
             'tag': self.tag.name,
             'id': self.id,
-            'name': self.name,
             'type': self.type.name,
             'perms': {k.name: v for k, v in self.perms.items()},
             'flags': {k.name: v for k, v in self.flags.items()},
             'text': self.text
         }
 
     def __setstate__(self, obj):
-        if 'text' in obj:
+        if 'text' in obj and obj['text'] is not None:
             self.text = obj['text']
 
-        if 'id' in obj:
-            self.id = obj['id']
+        if 'tag' in obj:
+            self.tag = obj['tag']
 
-        if 'name' in obj:
-            self.name = obj['name']
+        if 'id' in obj and obj['id'] is not None:
```

Comment:
Same here

Reply:
We need an explicit check for none type here. You can have id 0. bsd.acl sets "id = None" if ae_tag is ACL_USER_OBJ, ACL_GROUP_OBJ, or ACL_EVERYONE.

Reply:
`obj.get('id', None) is not None`, we can drop the first if as well ; )